### PR TITLE
[ADVAPI32] RegEnumKeyExW on HKCR keys must work with just KEY_ENUMERATE_SUB_KEYS access

### PR DESCRIPTION
**Pull request by whindsaks**
This fixes Windows RegEdit when the same HKCR key exists in HKCU and HKLM.

### CODE REVIEW

**Comment by JoachimHenze** [comment]
That was about the thing that regedit (ours and that of Windows) doesn't show any content, if the same key would exist in both, right?

**Comment by whindsaks** [comment]
> right

This fixes Windows Regedit. ROS Regedit is fixed by the other PR I made today.

**Review by JoachimHenze — APPROVED** [review]

File: dll/win32/advapi32/reg/hkcr.c [hunk 1]
```diff
@@ -214,6 +214,30 @@ GetPreferredHKCRKey(
     return ErrorCode;
 }
 
+static
+LONG
+GetSubkeyInfoHelper(
+    _In_ HKEY hKey,
+    _Out_ LPDWORD lpSubKeys,
+    _Out_ LPDWORD lpMaxSubKeyLen)
```
Comment (HBelusca) [inline dll/win32/advapi32/reg/hkcr.c:222]: From usage below, it appears those two last parameters would be `_Out_opt_`

File: dll/win32/advapi32/reg/hkcr.c [hunk 2]
```diff
@@ -214,6 +214,30 @@ GetPreferredHKCRKey(
...
+{
+    LONG err = RegQueryInfoKeyW(hKey, NULL, NULL, NULL, lpSubKeys, lpMaxSubKeyLen,
+                                NULL, NULL, NULL, NULL, NULL, NULL);
+    if (err != ERROR_ACCESS_DENIED)
+        return err;
+
+    /* Windows RegEdit only uses KEY_ENUMERATE_SUB_KEYS when enumerating but
+     * KEY_QUERY_VALUE is required to get the info in EnumHKCRKey.
+     */
+    if (!RegOpenKeyExW(hKey, NULL, 0, KEY_QUERY_VALUE, &hKey))
```
Comment (HBelusca) [inline dll/win32/advapi32/reg/hkcr.c:232]:
```suggestion
    if (RegOpenKeyExW(hKey, NULL, 0, KEY_QUERY_VALUE, &hKey) == ERROR_SUCCESS)
```

**Review by GeoB99 — APPROVED** [review]
LGTM

**Pull request merged**